app/src/main/java/protect/card_locker/ImportExportTask.java

```diff
@@ -87,17 +97,47 @@ private ImportExportResult performExport(Context context, OutputStream stream, S
     }
 
     public void onPreExecute() {
-        progress = new ProgressDialog(activity);
-        progress.setTitle(doImport ? R.string.importing : R.string.exporting);
+        int llPadding = 30;
+        LinearLayout ll = new LinearLayout(context);
+        ll.setOrientation(LinearLayout.HORIZONTAL);
+        ll.setPadding(llPadding, llPadding, llPadding, llPadding);
+        ll.setGravity(Gravity.CENTER);
+        LinearLayout.LayoutParams llParam = new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.WRAP_CONTENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT);
+        llParam.gravity = Gravity.CENTER;
+        ll.setLayoutParams(llParam);
+
+        ProgressBar progressBar = new ProgressBar(context);
+        progressBar.setIndeterminate(true);
+        progressBar.setPadding(0, 0, llPadding, 0);
+        progressBar.setLayoutParams(llParam);
+
+        llParam = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT,
+                ViewGroup.LayoutParams.WRAP_CONTENT);
+        llParam.gravity = Gravity.CENTER;
+        TextView tvText = new TextView(context);
+        tvText.setText(doImport ? R.string.importing : R.string.exporting);
+
+        int textColor;
+        if (Utils.isDarkModeEnabled(context)) {
+            textColor = Color.WHITE;
+        } else {
+            textColor = Color.BLACK;
+        }
+        tvText.setTextColor(textColor);
+        tvText.setTextSize(20);
+        tvText.setLayoutParams(llParam);
+
+        ll.addView(progressBar);
+        ll.addView(tvText);
 
-        progress.setOnDismissListener(new DialogInterface.OnDismissListener() {
-            @Override
-            public void onDismiss(DialogInterface dialog) {
-                ImportExportTask.this.stop();
-            }
-        });
+        AlertDialog.Builder builder = new MaterialAlertDialogBuilder(context);
+        builder.setCancelable(true);
+        builder.setView(ll);
 
-        progress.show();
+        dialog = builder.create();
+        dialog.show();
```

Comment:
It looks like you're fixing the ProgressDialog being deprecated here by recreating the ProgressDialog in a MaterialAlertDialog?

While that does remove the deprecation warning, it doesn't solve the real issue: importing is done on the main thread, locking up the UI and breaking if the user navigates away and Android stops the activity.

Because this doesn't fix the real issue and just adds a lot of extra code, I'd rather you just leave the ProgressDialog here and cause the in this case very correct deprecation warning to remain shown. #513 exists to track this specific issue.

Reply:
Reverted changes